src/main/java/at/ac/fhcampuswien/fhmdb/HomeController.java

```diff
@@ -113,4 +113,57 @@ private List<Integer> getYears() {
                 .mapToObj(i -> currentYear - i)
                 .collect(Collectors.toList());
     }
+
+
+    public String getMostPopularActor(List<Movie> movies) {
+        return movies.stream()
+                // convert every main cast into a separate stream and flatten the main cast of all movies into a single stream
+                .flatMap(movie -> movie.getMainCast().stream())
+
+                // Group actors by their name and count their occurrences using the Collectors.groupingBy method
+                .collect(Collectors.groupingBy(actor -> actor, Collectors.counting()))
+
+                // Create a stream of the entry set of the grouped actors
+                .entrySet().stream()
+
+                // Find the actor with the maximum count using the comparingByValue method of the Map.Entry
+                .max(Map.Entry.comparingByValue())
```

Comment:
If you have two entries with the same max value, it will return the last one, but I would rather compare them and return the first, so I would implement a custom comparator here, something like:
` .max((e1, e2) -> e1.getValue().equals(e2.getValue()) ? -1 : e1.getValue().compareTo(e2.getValue())`

Reply:
good point. Built it in